### PR TITLE
chore(duckdb): Refactor

### PR DESCRIPTION
**Pull request by candiduslynx**
It's an effort to shorten #11116 diff

### CODE REVIEW

File: plugins/destination/duckdb/client/types.go [hunk 1]
```diff
@@ -7,10 +7,45 @@ import (
...
+	fields := sc.Fields()
+	for i := range fields {
+		fields[i].Type = transformTypeForWriting(fields[i].Type)
+	}
+	md := sc.Metadata()
+	return arrow.NewSchema(fields, &md)
+}
+
+func transformType(dt arrow.DataType) arrow.DataType {
+	return duckDBToArrow(arrowToDuckDB(dt))
```
Comment (bbernays) [inline plugins/destination/duckdb/client/types.go:42]: Why wrap this function call in a function? What do we get for this added abstraction layer?

File: plugins/destination/duckdb/client/transform.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	for i := 0; i < int(rec.NumCols()); i++ {
+		cols[i] = transformArray(rec.Column(i))
+	}
+	return array.NewRecord(sc, cols, rec.NumRows())
+}
+
+func transformArray(arr arrow.Array) arrow.Array {
+	if plainWrite(arr.DataType()) {
+		return arr
+	}
```
Comment (bbernays) [inline plugins/destination/duckdb/client/transform.go:21]: Why wrap this single boolean expression in a function why not just directly call:
```suggestion
	if arrow.TypeEqual(arr.DataType(), transformTypeForWriting(arr.DataType())) {
		return arr
	}
```

File: plugins/destination/duckdb/client/types.go (hunk 1 above)
Comment (candiduslynx) [reply plugins/destination/duckdb/client/types.go]: Done in dafa00461042b75497856cabe705d2bc3c9d535e

File: plugins/destination/duckdb/client/transform.go (hunk 2 above)
Comment (candiduslynx) [reply plugins/destination/duckdb/client/transform.go]: Merged this + removed in dafa00461042b75497856cabe705d2bc3c9d535e

**Review by disq — APPROVED** [review]

**Pull request merged**